core/src/main/scala/kafka/server/AbstractFetcherManager.scala

```diff
@@ -32,6 +32,7 @@ abstract class AbstractFetcherManager[T <: AbstractFetcherThread](val name: Stri
   // package private for test
   private[server] val fetcherThreadMap = new mutable.HashMap[BrokerIdAndFetcherId, T]
   private val lock = new Object
+  @volatile
   private var numFetchersPerBroker = numFetchers
```

Comment:
Why should we add `volatile` to `numFetchersPerBroker`? I don't think it's related to this issue, right?